### PR TITLE
[ALLUXIO-3169] Log server graceful shutdown

### PR DESCRIPTION
**Pull request by aaudiber**
https://alluxio.atlassian.net/browse/ALLUXIO-3169

### CODE REVIEW

File: logserver/src/main/java/alluxio/logserver/AlluxioLogServerProcess.java [hunk 1]
```diff
@@ -135,8 +141,11 @@ public void stop() throws Exception {
         LOG.warn("Exception in closing server socket.", e);
       }
     }
-
-    mThreadPool.shutdown();
+    // Close all client sockets so that their serving threads can stop.
+    for (Socket socket : mClientSockets) {
+      socket.close();
```
[the remark this answers is not in the document]
Comment (aaudiber) [reply logserver/src/main/java/alluxio/logserver/AlluxioLogServerProcess.java]: good call, I've updated to use `synchronized`

File: logserver/src/main/java/alluxio/logserver/AlluxioLogServerProcess.java [hunk 2]
```diff
@@ -135,8 +150,15 @@ public void stop() throws Exception {
...
     }
-
-    mThreadPool.shutdown();
+    mThreadPool.shutdownNow();
+    // Close all client sockets so that their serving threads can stop. We shut down the threadpool
+    // before closing the sockets so that no new socket threads will be created after we try to
+    // close them all.
+    synchronized (mClientSockets) {
+      for (Socket socket : mClientSockets) {
+        socket.close();
```
Comment (tedyu) [inline logserver/src/main/java/alluxio/logserver/AlluxioLogServerProcess.java]:
```
    public synchronized void close() throws IOException {
```
I think the above call should be wrapped in try block.

File: logserver/src/main/java/alluxio/logserver/AlluxioLogServerProcess.java [hunk 3]
```diff
@@ -135,8 +150,15 @@ public void stop() throws Exception {
...
+      }
+    }
     boolean ret = mThreadPool.awaitTermination(THREAD_KEEP_ALIVE_TIME_MS, TimeUnit.MILLISECONDS);
```
Comment (tedyu) [inline logserver/src/main/java/alluxio/logserver/AlluxioLogServerProcess.java]: This seems ineffective given the added call to shutdownNow() on line 153

Comment (aaudiber) [reply logserver/src/main/java/alluxio/logserver/AlluxioLogServerProcess.java]: the javadoc for `shutdownNow` says to use `awaitTermination`

File: logserver/src/main/java/alluxio/logserver/AlluxioLogServerProcess.java (hunk 2 above)
Comment (aaudiber) [reply logserver/src/main/java/alluxio/logserver/AlluxioLogServerProcess.java]: Good call, created https://github.com/Alluxio/alluxio/pull/7225 to address this

**Pull request merged**